src/super_gradients/training/transforms/keypoints/keypoints_random_affine.py

```diff
@@ -0,0 +1,193 @@
+            f"max_scale={self.max_scale}, "
+            f"max_translate={self.max_translate}, "
+            f"image_pad_value={self.image_pad_value}, "
+            f"mask_pad_value={self.mask_pad_value}, "
+            f"prob={self.prob})"
+        )
+
+    def _get_affine_matrix(self, img, angle, scale, dx, dy):
+        """
+
+        :param center: (x,y)
+        :param scale:
+        :param output_size: (rows, cols)
+        :param rot:
+        :return:
+        """
+        height, width = img.shape[:2]
+        center = (width / 2 + dx * width, height / 2 + dy * height)
+        matrix = cv2.getRotationMatrix2D(center, angle, scale)
+
+        return matrix
+
+    def apply_to_sample(self, sample: PoseEstimationSample) -> PoseEstimationSample:
+        """
+
+        :param image: (np.ndarray) Image of shape [H,W,3]
+        :param mask: Single-element array with mask of [H,W] shape.
+        :param joints: Single-element array of joints of [Num instances, Num Joints, 3] shape. Semantics of last channel is: x, y, joint index (?)
+        :param area: Area each instance occipy: [Num instances, 1]
+        :return:
+        """
+
+        if random.random() < self.prob:
+            angle = random.uniform(-self.max_rotation, self.max_rotation)
+            scale = random.uniform(self.min_scale, self.max_scale)
+            dx = random.uniform(-self.max_translate, self.max_translate)
+            dy = random.uniform(-self.max_translate, self.max_translate)
+            interpolation = random.choice(self.interpolation_mode)
+
+            mat_output = self._get_affine_matrix(sample.image, angle, scale, dx, dy)
+            mat_output = mat_output[:2]
+
+            image_pad_value = (
+                tuple(self.image_pad_value) if isinstance(self.image_pad_value, Iterable) else tuple([self.image_pad_value] * sample.image.shape[-1])
+            )
+
+            sample.image = self.apply_to_image(sample.image, mat_output, interpolation, image_pad_value, cv2.BORDER_CONSTANT)
+            sample.mask = self.apply_to_image(sample.mask, mat_output, cv2.INTER_NEAREST, self.mask_pad_value, cv2.BORDER_CONSTANT)
+
+            new_image_shape = sample.image.shape
+            sample.joints = self.apply_to_keypoints(sample.joints, mat_output, new_image_shape)
+
+            if sample.bboxes is not None:
+                sample.bboxes = self.apply_to_bboxes(sample.bboxes, mat_output)
+
+            if sample.areas is not None:
+                sample.areas = self.apply_to_areas(sample.areas, mat_output)
+
+            sample = sample.sanitize_sample()
+
+        return sample
+
+    @classmethod
+    def apply_to_areas(cls, areas: np.ndarray, mat):
+        det = np.linalg.det(mat[:2, :2])
+        return (areas * abs(det)).astype(areas.dtype)
+
+    @classmethod
+    def apply_to_bboxes(cls, bboxes_xywh: np.ndarray, mat: np.ndarray):
+        """
+
+        :param bboxes: (N,4) array of bboxes in XYWH format
+        :param mat:
```

Comment:
same, docstring for mat

Reply:
Fixed in 88cef73f0a3c26f9e158589c0d36ef541787eaee